rfp-proposals/rfp-proposal-reputation-system.md

```diff
@@ -0,0 +1,100 @@
+# RFP Proposal: `Reputation System`
+
+**Name of Project:** Reputation System
+
+**Link to RFP:** [Reputation System](https://github.com/filecoin-project/devgrants/blob/master/rfps/rfp-reputation-system.md)
+`
+**RFP Category:** `app-dev`
+
+**Proposer:** [@deaswang](https://github.com/deaswang)
+
+**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and APACHE2 licenses?:** "Yes"
+
+# Project Description
+
+a reputation system include:
+
+- web site:
+  - show list of miners order by its reputation.
+  - miner detail information include:
+    - Network participants should be able to query available miners and get back a response that includes (but not limited to): 
+    - Number and % (where applicable) of deals with no penalties
+    - Number and % (where applicable) of slashed contracts
+    - Number and % (where applicable) of dropped contracts
+    - Number and % of failed deals (if possible) -- i.e. the number of deals that were attempted but failed, possibly due to this miner having spotty connectivity
+    - % of retrieval requests successfully fulfilled
+    - Number of files retrieved
+    - Lifetime data stored
+    - Lifetime data retrieved
+    - etc
+  - Storage client Feedback.
```

Comment:
How will storage clients provide this feedback?

Reply:
we can get which client store data to the miner from chain and use client private key to sign feedback and verify it.